docs/madr/decisions/046-policy-on-namespace.md

```diff
@@ -0,0 +1,456 @@
+              - kind: MeshServiceSubset
+                name: backend
+                tags:
+                  version: v2
+---
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: backend-route-timeout
+  namespace: kuma-system
+spec:
+  targetRef:
+    kind: MeshHTTPRoute
+    name: backend-route
+  to:
+    - targetRef:
+        kind: Mesh
+      default:
+        http:
+          requestTimeout: 5s
+# on frontend-ns
+---
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: frontend-to-backend
+  namespace: frontend-ns
+spec:
+  targetRef:
+    kind: MeshHTTPRoute
+    name: backend-route
+    namespace: kuma-system
+  to:
+    - targetRef:
+        kind: Mesh
+      default:
+        http:
+          requestTimeout: 3s
+```
+
+This example also demonstrates the scenario when 2 MeshTimeout reference the same route,
+the namespaced MeshTimeout is considered more specific.
+That's why the final request timeout value should be 3s.
+
+**4. Both MeshTimeout and MeshHTTPRoute in "frontend-ns"**
+
+This scenario is likely to be the most common way users create policies.
+The use case for this is:
+
+**As** a Frontend Service Owner<br>
+**I want** all outgoing requests from fronted (to backend) to have headers starting with `xxx-fronted-` removed<br>
+**so that** requests are smaller and backend doesn't receive unnecessary data.
+
+**As** a Frontend Service Owner<br>
+**I want** all requests routed to `backend` to have 3s timeout<br>
+**so that** my service meets SLO and responds withing 3s.
+
+This can be achieved by applying 2 policies:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshHTTPRoute
+metadata:
+  name: backend-route
+  namespace: frontend-ns
+spec:
+  targetRef:
+    kind: MeshService
+    name: frontend
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+      rules:
+        - matches:
+            - path:
+                type: PathPrefix
+                value: /
+          default:
+            filters:
+              - type: RequestHeaderModifier
+                requestHeaderModifier:
+                  remove:
+                    - xxx-frontend-version
+                    - xxx-frontend-data
+                    - xxx-frontend-content
+            backendRefs:
+              - kind: MeshService
+                name: backend
+---
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: frontend-to-backend
+  namespace: frontend-ns
+spec:
+  targetRef:
+    kind: MeshHTTPRoute
+    name: backend-route
+    namespace: frontend-ns # can be omitted as it's a default value
+  to:
+    - targetRef:
+        kind: Mesh
+      default:
+        http:
+          requestTimeout: 3s
+```
+
+**5. MeshTimeout in "frontend-ns", MeshHTTPRoute in "backend-ns"**
+
+MeshHTTPRoute in `backend-ns` produces envoy routes in DPPs in `backend-ns`.
+Ability to configure timeouts on these routes from the `frontend-ns` breaks the namespace isolation and seems
```

Comment:
I think this is my ideal workflow for implementing 3., where `backend-ns` has some authority over requests to its service from anywhere in the mesh and `frontend-ns` can override that for its own Services.

Reply:
It will be like that after we updated MADR